x-pack/plugin/esql/src/main/generated/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvAppendBooleanEvaluator.java

```diff
@@ -41,40 +43,48 @@ public MvAppendBooleanEvaluator(Source source, EvalOperator.ExpressionEvaluator
   @Override
   public Block eval(Page page) {
     try (BooleanBlock field1Block = (BooleanBlock) field1.eval(page)) {
-      try (BooleanBlock field2Block = (BooleanBlock) field2.eval(page)) {
-        return eval(page.getPositionCount(), field1Block, field2Block);
+      BooleanBlock[] field2Blocks = new BooleanBlock[field2.length];
+      try (Releasable field2Release = Releasables.wrap(field2Blocks)) {
+        for (int i = 0; i < field2Blocks.length; i++) {
+          field2Blocks[i] = (BooleanBlock)field2[i].eval(page);
+        }
+        return eval(page.getPositionCount(), field1Block, field2Blocks);
       }
     }
   }
 
-  public BooleanBlock eval(int positionCount, BooleanBlock field1Block, BooleanBlock field2Block) {
+  public BooleanBlock eval(int positionCount, BooleanBlock field1Block,
+      BooleanBlock[] field2Blocks) {
     try(BooleanBlock.Builder result = driverContext.blockFactory().newBooleanBlockBuilder(positionCount)) {
+      BooleanBlock[] field2Values = new BooleanBlock[field2.length];
       position: for (int p = 0; p < positionCount; p++) {
         boolean allBlocksAreNulls = true;
         if (!field1Block.isNull(p)) {
           allBlocksAreNulls = false;
         }
-        if (!field2Block.isNull(p)) {
-          allBlocksAreNulls = false;
-        }
         if (allBlocksAreNulls) {
           result.appendNull();
           continue position;
         }
-        MvAppend.process(result, p, field1Block, field2Block);
+        // unpack field2Blocks into field2Values
+        for (int i = 0; i < field2Blocks.length; i++) {
+          int o = field2Blocks[i].getFirstValueIndex(p);
+          field2Values[i] = field2Blocks[i];
```

Comment:
It's unclear to me why we need to copy the `field2Blocks` into another `BooleanBlock[]`; this seems to serve no purpose.

Reply:
Yes, it is redundant operations, I modified the logic during code generation to skip generating this copy operation.